lib/tasks/scm.rake

```diff
@@ -0,0 +1,43 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+namespace :scm do
+  namespace :migrate do
+    desc 'Migrate existing repositories to managed for a given URL prefix'
+    task managed: :environment do |task, args|
+
+      urls = args.extras
+      abort "Requires at least one URL prefix to identify existing repositories" if urls.length < 1
+
+      urls.each do |url|
+        Repository.where('url LIKE ?', "#{url}%").update_all(scm_type: :managed)
```

Comment:
Won't this escape your `%`, because it is part of the untrusted input?

Shouldn't this read like: `where('url LIKE ?%', url)`?